src/event/mod.rs

```diff
@@ -159,6 +158,37 @@ impl<K: Into<Atom>, V: Into<Value>> FromIterator<(K, V)> for LogEvent {
     }
 }
 
+pub fn log_schema<'a>() -> &'a LogSchema {
+    // TODO: Help Rust project support before_each
+    // Support uninitialized schemas in tests to help our contributors.
+    // Don't do it in release because that is scary.
+    #[cfg(debug_assertions)]
+    {
+        if LOG_SCHEMA.get().is_none() {
+            error!("You are not initializing a schema in this test -- This could fail in release");
+            LOG_SCHEMA.set(LogSchema::default()).ok(); // If this fails it means some other test set it while we were trying to.
+        }
+    }
+    LOG_SCHEMA.get().expect("Schema was not initialized")
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct LogSchema {
+    pub message_key: Atom,
```

Comment:
I would rather prefer we made these getter fns rather than pub struct items, what do you think?

Reply:
🤔 I had these private, but a test wanted them public, let me revisit it.